controllers/kmsissuer_controller.go

```diff
@@ -138,23 +141,28 @@ func ParseCertificate(cert []byte) (*x509.Certificate, error) {
 }
 
 // certificateNeedsRenewal returns True if the certificate needs to be created/renewed.
-func certificateNeedsRenewal(issuer *kmsiapi.KMSIssuer, desiredCert *x509.Certificate) bool {
+func (r *KMSIssuerReconciler) certificateNeedsRenewal(issuer *kmsiapi.KMSIssuer, desiredCert *x509.Certificate) bool {
+	log := r.Log.WithValues("name", issuer.Name, "namespace", issuer.Namespace)
 	// Check if the certificate hasn't been issued yet.
 	if len(issuer.Status.Certificate) == 0 {
+		log.Info("certificate hasn't been issued yet")
 		return true
 	}
 	// Check if the existing cetificate is valid.
 	actualCert, err := ParseCertificate(issuer.Status.Certificate)
 	if err != nil {
+		log.Info("existing certificate isn't valid", "error", err)
```

Comment:
Why do you pass some fields with `WithValues` and some in `Info`?

Reply:
`WithValues` configures the logger. `Info` emits a log entry.
https://github.com/go-logr/logr/blob/master/logr.go#L47-L50